### PR TITLE
CMR-5071 As a Client User, I want to parse and generate two-d-coordinate-system in SWOT granule metadata format

### PR DESCRIPTION
**Pull request by daniel-zamora**
This PR includes some refactoring of the umm-g test namespace, and the mappings TilingIdentificationSystem <=> TwoDCoordinateSystem for UMM-G v1.4

### CODE REVIEW

File: umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj [hunk 1]
```diff
@@ -57,19 +58,19 @@
...
       :temporal (umm-g->Temporal umm-g-json)
-      ; :orbit-calculated-spatial-domains (ocsd/xml-elem->orbit-calculated-spatial-domains umm-g-json)
       :platform-refs (platform/umm-g-platforms->PlatformRefs (:Platforms umm-g-json))
       :project-refs (project/umm-g-projects->ProjectRefs (:Projects umm-g-json))
       :cloud-cover (:CloudCover umm-g-json)
-      ; :two-d-coordinate-system (two-d/xml-elem->TwoDCoordinateSystem umm-g-json)
-      ; :related-urls (ru/xml-elem->related-urls umm-g-json)
-      ; :spatial-coverage (xml-elem->SpatialCoverage umm-g-json)
+      :two-d-coordinate-system (tiling-system/umm-g-tiling-identification-system->TwoDCoordinateSystem
+                                (:TilingIdentificationSystem umm-g-json))})))
```
Comment (ygliuvt) [inline umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj]: Please do not move around the lines or parentheses. Just uncomment out the field that you are changing to minimize the merge conflict with other people also making changes to UMM-G.

File: umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj [hunk 2]
```diff
@@ -105,4 +106,5 @@
      :Platforms (platform/PlatformRefs->umm-g-platforms platform-refs)
      :CloudCover cloud-cover
      :Projects (project/ProjectRefs->umm-g-projects project-refs)
-     }))
+     :TilingIdentificationSystem (tiling-system/TwoDCoordinateSystem->umm-g-tiling-identification-system
+                                  two-d-coordinate-system)}))
```
Comment (ygliuvt) [inline umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj]: Please do not move around the lines or parentheses. Just uncomment out the field that you are changing to minimize the merge conflict with other people who are also making changes to UMM-G.

File: umm-spec-lib/src/cmr/umm_spec/umm_g/tiling_system.clj [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+         :start-coordinate-2 (:MinimumValue Coordinate2)
+         :end-coordinate-2 (:MaximumValue Coordinate2)}))))
+
+(defn TwoDCoordinateSystem->umm-g-tiling-identification-system
+  "Returns the UMM-G TilingIdentificationSystem from the given umm-lib granule model TwoDCoordinateSystem."
+  [two-d-coordinate-system]
+  (when two-d-coordinate-system
+    (let [{:keys [name start-coordinate-1 end-coordinate-1 start-coordinate-2 end-coordinate-2]}
+          two-d-coordinate-system]
+      {:TilingIdentificationSystemName name
```
Comment (ygliuvt) [inline umm-spec-lib/src/cmr/umm_spec/umm_g/tiling_system.clj]: What happens when the umm-lib TwoDCoordinateSystemName is not in the defined ENUMs of UMM-G?

File: umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj (hunk 2 above)
Comment (daniel-zamora) [reply umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj]: I think we should maintain proper clojure coding style with respect to parenthesis and spacing.

File: umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj (hunk 1 above)
Comment (daniel-zamora) [reply umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj]: Let me change it back to the order it was then.

File: umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj (hunk 2 above)
Comment (ygliuvt) [reply umm-spec-lib/src/cmr/umm_spec/umm_g/granule.clj]: This is a temporary thing. Once we added all the fields, we can do the proper formatting. 

Or a better way is if you make a version that works with parinfer and with the commented lines in between fixed lines (to minimize merge conflict), that would be great. I will go with that.

File: umm-spec-lib/src/cmr/umm_spec/umm_g/tiling_system.clj (hunk 3 above)
Comment (daniel-zamora) [reply umm-spec-lib/src/cmr/umm_spec/umm_g/tiling_system.clj]: when/if it is validated against the umm-g json schema, there will be an error.

Comment (ygliuvt) [reply umm-spec-lib/src/cmr/umm_spec/umm_g/tiling_system.clj]: I think we want to map the name to a default value in the defined ENUM list. A valid ECHO10 granule with TwoDCoordinateSystem should not become invalid when it is converted into UMM-G format.

**Pull request merged**